vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/ClientTypes.java

```diff
@@ -198,4 +199,48 @@ static public int mapDB2TypeToDriverType(boolean isDescribed, int sqlType, long
             return 0;
         }
     }
+    
+    public static boolean canConvert(Object value, int toType) {
+    	Class<?> clazz = value.getClass();
+    	// Everything can convert to String
+    	if (clazz == String.class)
+    		return true; 
+    	switch (toType) {
+        case ClientTypes.BIGINT:
+        case ClientTypes.BIT:
+        case ClientTypes.BOOLEAN:
+        case ClientTypes.CHAR:
+        case ClientTypes.DECIMAL:
+        case ClientTypes.DOUBLE:
+        case ClientTypes.INTEGER:
+        case ClientTypes.LONGVARCHAR:
+        case ClientTypes.REAL:
+        case ClientTypes.SMALLINT:
+        	return clazz == boolean.class ||
+        		   clazz == Boolean.class ||
+        		   clazz == double.class ||
+        		   clazz == Double.class ||
+        		   clazz == float.class ||
+        		   clazz == Float.class ||
+        		   clazz == int.class ||
+        		   clazz == Integer.class ||
+        		   clazz == long.class ||
+        		   clazz == Long.class ||
+        		   clazz == short.class ||
+        		   clazz == Short.class ||
+        		   clazz == BigDecimal.class;
+        case ClientTypes.BINARY:
+        case ClientTypes.BLOB:
+        	return clazz == boolean.class ||
+        		   clazz == Boolean.class ||
```

Comment:
For BINARY and BLOB, why do we return true if it is a B/boolean?

Reply:
According to[ this DB2 doc](https://www.ibm.com/support/knowledgecenter/SSEPEK_12.0.0/intro/src/tpc/db2z_stringdatatypes.html) a BINARY column is a varying-length binary string where `0 < length <= 255` with a default length of 1. I think the most common use case here is to just store a length=1 binary string which would fit into a B/boolean Java type.

A BLOB is also a variable-length binary string where `0 <= length <= 2G` with a default size of 1M. Normally this will only fit into a byte array, but in some cases it could be length=1.

Overall, the checks in this method don't need to be perfect, they are just client-side checking and if we get some incompatible data the user will get an error message later on, but the error message we can give here will probably be more specific/clear than the error message later on.